core/trino-main/src/main/java/io/trino/sql/analyzer/FeaturesConfig.java

```diff
@@ -80,6 +80,7 @@
     private int maxReorderedJoins = 9;
     private boolean redistributeWrites = true;
     private boolean usePreferredWritePartitioning;
+    private int preferredWritePartitioningMinNumberOfPartitions;
```

Comment:
I suggest to set it to some default value, e.g: `50`. This should:
* make preferred partitioning to be used when there is full table CTAS with more than 50 partitions
* provide good concurrency (>50) for preferred partitioning
* mitigate hive writer partition limit (100)
* provide good concurrency for inserts into single partition

WDYT: @findepi @willmostly ?

Reply:
Please set it to 50.